include/onnx-mlir/Runtime/OMEntryPoint.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===----------- OMEntryPoint.h - OMEntryPoint Declaration header ---------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains declaration of OMEntryPoint API functions.
+//
+//===----------------------------------------------------------------------===//
+
+#ifdef __cplusplus
+#pragma once
+
+extern "C" {
+#endif
+
+/**
+ * \brief Return all entry point names in a model.
+ *
+ * @return an array of strings. The array ends with NULL. For example:
+ * ["run_add", "run_sub", NULL].
+ */
+const char **omQueryEntryPoints();
```

Comment:
NULL terminated string array is fairly common. Unless there are many use cases where you only want the number of entry points but not the entry points themselves, this would be the simplest API.

Reply:
It is fine, but the type may need to be more `const char * const *` as you have pointer to a constant array of pointers, each of which points to a constant char string.

Reply:
Done. I changed to `const char * const *`. Keep using NULL-terminated arrays. Thanks!